src/ucm/util/reloc.c

```diff
@@ -312,12 +312,52 @@ static ucs_status_t ucm_reloc_apply_patch(ucm_reloc_patch_t *patch,
     return ctx.status;
 }
 
+/* read serinfo from 'module_path', result buffer must be destroyed
+ * by free() call */
+static Dl_serinfo *ucm_dlopen_load_serinfo(const char *module_path)
+{
+    Dl_serinfo *serinfo = NULL;
+    Dl_serinfo stub;
+    void *caller_module;
+    int res;
+
+    caller_module = ucm_reloc_orig_dlopen(module_path, RTLD_NOW);
+    if (caller_module == NULL) { /* caller module can't be loaded */
+        ucm_debug("failed to open %s: %s", module_path, dlerror());
+        return NULL;
+    }
+
+    /* try to get search info from caller module */
+    res = dlinfo(caller_module, RTLD_DI_SERINFOSIZE, &stub);
+    if (res) {
+        ucm_debug("dlinfo(RTLD_DI_SERINFOSIZE) failed");
+        goto close_caller;
+    }
+
+    serinfo = malloc(stub.dls_size);
+    if (serinfo == NULL) {
+        ucm_error("failed to allocate %zu bytes", stub.dls_size);
```

Comment:
"failed to allocate %zu bytes for Dl_serinfo"

Reply:
string updated